lib/sprockets/cached_digest_utils.rb

```diff
@@ -0,0 +1,17 @@
+module Sprockets
+  module CachedDigestUtils
+    extend self
+
+    def cached_version_digest
+      version_string = defined?(version) ? version : ''
+
+      # Compute the initial digest using the implementation class. The
+      # Sprockets release version and custom environment version are
+      # mixed in. So any new releases will affect all your assets.
+      @cached_version_digest ||= digest_class.new.update(version_string.to_s)
+
+      # Returned a dupped copy so the caller can safely mutate it with `.update`
+      @cached_version_digest.dup
+    end
+  end
+end
```

Comment:
This is actually about the third iteration of the changes in this one PR which might be partially to blame for the confusion :P

Initially, this started out as just a straight port of #404 but that broke the integrity attribute we were using for subresource integrity. I played around with a few things, rebased this branch to remove all the guff and then realised the version wasn't being incremented for all assets. I finally managed to stumble onto sellect/sprockets#1 which correctly solved the integrity issue so I rebased that in but now the JS filenames are including the version so I'm back to investigating that.

For now, it's probably safe to ignore this. I might even end up closing this until I solve the underlying issue here and work out if it's in sprockets or our app.